app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFragment.kt

```diff
@@ -156,12 +133,14 @@ private fun ContentView(
     topBar = { NavigationBar(onFilter, onSearch) },
     backgroundColor = AlgoismeTheme.colors.primaryVariant
 ) {
+    if (problemsState.value == null) return@Scaffold
```

Comment:
`val state = problemsStates.value ?: return@Scaffold` will allow you to have non-optional state further.